lib/ansible/module_utils/basic.py

```diff
@@ -1255,7 +1280,7 @@ def _log_invocation(self):
             msg = 'Invoked'
 
         # 6655 - allow for accented characters
-        if isinstance(msg, unicode):
+        if not isinstance(msg, str):
             # We should never get here as msg should be type str, not unicode
             msg = msg.encode('utf-8')
```

Comment:
* I wonder if we need to also catch things that are not text (for instance a number)?
* The previous code handled that further down with a format string.  We also wouldn't want to throw a traceback here on non-utf8 bytes.  Since this is logging we probably want to use errors='replace' to deal with that.
* I took a look at systemd.journal on fedora21.  (Note: these behaviours are unaffected by client locale settings.  Don't know if the systemd daemon has a settable locale setting).
  * in python2 it will take either bytes or unicode string.  If the string is not utf-8, systemd.journal registers it as a binary blob.
  * On python3 this must be a unicode string.
* Also took a look at syslog on fedora 21(these are unaffected by the client locale setting)
  * In python2 it must be a byte string.  A non-utf-8 string becomes the replacement character
  * In python3 this must be a text str type.

So this whole method needs to be reworked for python3 compatibility.